detect_secrets/core/usage.py

```diff
@@ -120,7 +120,7 @@ def add_arguments(self):
     def _add_initialize_baseline_argument(self):
         self.parser.add_argument(
             'path',
-            nargs='?',
+            nargs='*',
```

Comment:
I'm not sure, but would it be possible to use `'+'`? So you wouldn't have to do
```python
:type path: str|list
```
and the `isinstance` check

Reply:
Well, as '+' generates an error message if there isn't at least one command-line argument present... I thought that as we are using default's, '*' would be a better fit. Running the test suite with '+' breaks with `detect-secrets scan: error: too few arguments` (Well, as intended!).

Maybe I'm missing something... But If you think it's an improvement I can take a closer look at it :)

Reply:
You’re completely right :) 

From reading https://stackoverflow.com/questions/23490152/list-of-arguments-with-argparse#23490179 it seems it should be always be a list with *, so maybe that’s true?

Reply:
Yes! Looks like all is passed as a list, even the default argument ` '.' `

So the isInstance() might be redundant as when we pass (on the initialize()) `path='.' `  it's never used... Running scan without path(s) will trigger argparse's default. 

This was closely related to https://github.com/Yelp/detect-secrets/pull/188#discussion_r288614772 (@domanchi) so maybe using the list as a default isn't that great... But I think '+' passes all as a list too! What's the best way to handle this?

Reply:
If I understand, the 

```python
    if not isinstance(path, list):
        path = [path]
```
is always dead-code, so we can remove that. 

From reading the code, it seems a path is always passed in to `initialize`, so there doesn't seem to be a need for a default argument I think, right?

If so we can change
```python
def initialize(
    plugins,
    exclude_files_regex=None,
    exclude_lines_regex=None,
    path='.',
    scan_all_files=False,
):
```
to
```
def initialize(
    plugins,
    path,
    exclude_files_regex=None,
    exclude_lines_regex=None,
    scan_all_files=False,
):
```

Reply:
As I thought :) Thanks for the writeup. Updated the PR and fixed the failing tests.

Reply:
Looks awesome 👏